src-tauri/src/main.rs

```diff
@@ -58,19 +59,22 @@ fn main() {
     gather_parent_env_vars(&mut engine_state, &std::env::current_dir().unwrap());
 
     let stack = Stack::new();
+    let card_cache = HashMap::<String, nu_protocol::Value>::new();
 
     #[cfg(target_os = "macos")]
     {
         tauri::Builder::default()
             .manage(NanaState {
                 engine_state: Mutex::new(engine_state),
                 stack: Mutex::new(stack),
+                card_cache: Mutex::new(card_cache),
```

Comment:
Why does the state need a mutex around each property?

Reply:
In case multiple commands try to modify state simultaneously.

I'm not 100% clear on Tauri's threading model here (because it isn't documented), and eventually I need to read through the source to figure it out. But for now, a mutex is a pretty easy way to cover our butts.

Reply:
> In case multiple commands try to modify state simultaneously.
> 
> I'm not 100% clear on Tauri's threading model here (because it isn't documented), and eventually I need to read through the source to figure it out. But for now, a mutex is a pretty easy way to cover our butts.

Tauris threading model is the multithreaded model from tokio